server/storage/backend/batch_tx.go

```diff
@@ -133,6 +150,7 @@ func (t *batchTx) UnsafeDeleteBucket(bucket Bucket) {
 		)
 	}
 	t.pending++
+	t.pendingDeleteOperations++
```

Comment:
From what I remember bucket operations are not buffered directly, but somewhat implicitly. Will just tracking pending deletions work for them? Have you tested this? Could you add a test for `UnsafeDeleteBucket`?

Reply:
added a test for `UnsafeDeleteBucket`. It would fail without the fix.